src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1774,13 +1782,31 @@ else if (con.getTrustManagerClass() != null) {
             if (logger.isLoggable(Level.FINEST))
                 logger.finest(toString() + " Getting TLS or better SSL context");
 
-            sslContext = SSLContext.getInstance(sslProtocol);
-            sslContextProvider = sslContext.getProvider();
+            if (null != clientCertificate) {
+                try {
+                    KeyManager[] km = SQLServerCertificateUtils.getKeyManagerFromFile(clientCertificate, clientKey,
+                            clientKeyPassword);
 
-            if (logger.isLoggable(Level.FINEST))
-                logger.finest(toString() + " Initializing SSL context");
+                    sslContext = SSLContext.getInstance(sslProtocol);
+                    sslContextProvider = sslContext.getProvider();
+
+                    if (logger.isLoggable(Level.FINEST))
+                        logger.finest(toString() + " Initializing SSL context");
+
+                    sslContext.init(km, tm, null);
+                } catch (NullPointerException | FileNotFoundException e) {
```

Comment:
Catching NullPointerException ???

Reply:
We forgot to remove this exception, I've added null checks in the finally blocks inside the SQLServerCertificateUtils.java. We shouldn't need to catch NPE.